ui/service.go

```diff
@@ -35,7 +35,14 @@ func newServiceView(services []types.Service, app *App) *ServiceView {
 	}
 }
 
-func (app *App) showServicesPage() error {
+func (app *App) showServicesPage(refreshData bool) error {
+	pageName := ServicePage.getAppPageName(*app.cluster.ClusterArn)
+	if app.Pages.HasPage(pageName) && !refreshData {
+		app.Pages.SwitchToPage(pageName)
+
+		return nil
+	}
```

Comment:
☝️ If the cluster's services page was previously rendered, recycle that previous rendering of the page. This cuts out the time it would otherwise take to fetch data via the AWS API and render the results, which can be ~5 seconds when there are 400-500 services in my experience.